### PR TITLE
Enable dexcount plugin

### PR DESCRIPTION
**Pull request by serj-lotutovici**
Count methods on each build with -PwithDexcount

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
> Process 'Gradle Test Executor 2' finished with non-zero exit value 137

Not enough memory to do the build.

I have a branch where I divide execution of build steps to reduce memory usage, u can take a look at it here https://github.com/artem-zinnatullin/qualitymatters/commit/9322fd1fc93c2750354b30bf94f021ae92dd8919

File: app/build.gradle [hunk 1]
```diff
@@ -181,3 +181,17 @@ configurations.all {
...
     }
 }
+
+// Check if the build requires dex count
+def withDexcount = project.hasProperty('withDexcount')
+if (withDexcount) {
+    apply plugin: 'com.getkeepsafe.dexcount'
+
+    dexcount {
+        format = "list"
```
Comment (artem-zinnatullin) [inline app/build.gradle]: single quotes

File: app/build.gradle [hunk 2]
```diff
@@ -181,3 +181,17 @@ configurations.all {
         force libraries.supportAnnotations
     }
 }
+
+// Check if the build requires dex count
+def withDexcount = project.hasProperty('withDexcount')
+if (withDexcount) {
```
Comment (artem-zinnatullin) [inline app/build.gradle]: I like that you apply plugin only if it's needed, 👍

**Comment by artem-zinnatullin** [comment]
👍

**Comment by artem-zinnatullin** [comment]
@vanniktech, @lenguyenthanh PTAL

**Comment by serj-lotutovici** [comment]
Btw, the current method count is:
```
Total methods in app-1.0-1-debug.apk: 49745 (75.91% used)
Total fields in app-1.0-1-debug.apk:  25691 (39.20% used)
Methods remaining in app-1.0-1-debug.apk: 15790
Fields remaining in app-1.0-1-debug.apk:  39844
```
I'm still looking for a way to publish the values on github pr, using travis or any other ci. At my company we are using Jenkins, and there are a few plugins for that.

File: app/build.gradle (hunk 2 above)
Comment (serj-lotutovici) [inline app/build.gradle]: Saves build time, but not much.

**Comment by vanniktech** [comment]
:+1:

**Comment by lenguyenthanh** [comment]
👍 this plugin is really cool.

**Comment by artem-zinnatullin** [comment]
Thanks, @serj-lotutovici!

**Pull request merged**